source/geometries/Next100SiPM.cc

```diff
@@ -0,0 +1,191 @@
+                                1.99443901 * eV, 2.05580636 * eV, 2.12107005 * eV, 2.17476803 * eV, 2.23125551 * eV,
+                                2.29951572 * eV, 2.37208426 * eV, 2.46950045 * eV, 2.75730000 * eV, 3.04467524 * eV,
+                                3.14006977 * eV, 3.20705792 * eV, 3.2592052  * eV, 3.31307637 * eV, 3.38773724 * eV,
+                                3.44597914 * eV, 3.50625866 * eV, 3.54763044 * eV, 3.58999021 * eV, 3.61155171 * eV,
+                                3.65546116 * eV, 3.67781872 * eV, 3.72336446 * eV, 3.74656299 * eV, 3.79383824 * eV,
+                                3.86703126 * eV, 4.0 * eV, 4.2 * eV, 11.5 * eV};
+
+  G4double reflectivity[entries] = {0.0,  0.0, 0.0, 0.0, 0.0,
+                                    0.0,  0.0, 0.0, 0.0, 0.0,
+                                    0.0,  0.0, 0.0, 0.0, 0.0,
+                                    0.0,  0.0, 0.0, 0.0, 0.0,
+                                    0.0,  0.0, 0.0, 0.0, 0.0,
+                                    0.0,  0.0, 0.0, 0.0, 0.0,
+                                    0.0,  0.0, 0.0, 0.0, 0.0,
+                                    0.0, 0.0, 0.0, 0.0, 0.0,
+                                    0.0, 0.0};
+
+  G4double efficiency[entries]   = {0.0, 0.0, 0.0,
+                                    0.03506431, 0.0560223 , 0.07944598, 0.10291004, 0.12607905,
+                                    0.14954693, 0.17061431, 0.193974  , 0.21740781, 0.23843982,
+                                    0.26178898, 0.28501443, 0.30860135, 0.33010702, 0.35361856,
+                                    0.37751412, 0.40109223, 0.42743901, 0.43779628, 0.41021884,
+                                    0.38111716, 0.36057765, 0.33517116, 0.31491576, 0.28931617,
+                                    0.26968918, 0.24455707, 0.22893024, 0.20156831, 0.18538152,
+                                    0.15444223, 0.12549548, 0.1040183 , 0.0902669 , 0.05498552,
+                                    0.02944706, 0.0, 0.0, 0.0};
+
+  G4MaterialPropertiesTable* sipm_mt = new G4MaterialPropertiesTable();
+  sipm_mt->AddProperty("EFFICIENCY", energies, efficiency, entries);
+  sipm_mt->AddProperty("REFLECTIVITY", energies, reflectivity, entries);
+  G4OpticalSurface* sipm_opsurf =
+    new G4OpticalSurface("SIPM_OPSURF", unified, polished, dielectric_metal);
+  sipm_opsurf->SetMaterialPropertiesTable(sipm_mt);
+  new G4LogicalSkinSurface("SIPM_OPSURF", sens_logic_vol, sipm_opsurf);
+
+  // SENSITIVE DETECTOR ////////////////////////////////////
+
+  G4String sdname = "/" + sipm_name + "/";
+  G4SDManager* sdmgr = G4SDManager::GetSDMpointer();
+
+  if (!sdmgr->FindSensitiveDetector(sdname, false)) {
+    PmtSD* sensdet = new PmtSD(sdname);
+
+    if (sensor_depth_ == -1)
+      G4Exception("[Next100SiPM]", "Construct()", FatalException,
+                  "Sensor Depth must be set before constructing");
+    sensdet->SetDetectorVolumeDepth(sensor_depth_);
+
+    if (mother_depth_ == -1)
```

Comment:
This `if` clause and the following one is not necessary, because the default value is `0`, not `-1`. We set them to `0` because one could use this SiPM directly inside the gas or a support, and not in a dice board, which is replicated. `mother_depth` and `naming_order` are only needed in the latter kind of situation, otherwise they can be left at their default value (0).

Reply:
You're right. I started setting `mother_depth` and `naming_order` to `-1`, but later on I set them to `0`. I'll remove them.